projects/validation-messages/src/lib/components/validation-messages/validation-messages.component.spec.ts

```diff
@@ -1,44 +1,138 @@
 
 describe('ValidationMessagesComponent', () => {
-  let component: ValidationMessagesComponent;
-  let fixture: ComponentFixture<ValidationMessagesComponent>;
+  let spectator: SpectatorHost<ValidationMessagesComponent>;
+
+  const createHost = createHostFactory({
+    component: ValidationMessagesComponent,
+    componentProviders: [MatFormField],
+    providers: [ValidationMessagesService, ControlContainer],
+  });
+
+  const hostSetup = (
+    control?: FormControl,
+    controlName?: string,
+    errorMessages?: ValidationMessagesConfig,
+    apiErrorMessages?: ApiErrorMessages,
+    multiple = false
+  ) => {
+    return createHost(
+      `<ng-validation-messages
+    [control]="control"
+    [controlName]="controlName"
+    [errorsMessages]="errorMessages"
+    [apiErrorMessages]="apiErrorMessages"
+    [multiple]="multiple"></ng-validation-messages>`,
+      {
+        hostProps: {
+          control: control,
+          controlName: controlName,
+          errorsMessages: errorMessages,
+          apiErrorMessages: apiErrorMessages,
+          multiple: multiple,
+        },
+      }
+    );
+  };
+
+  const apiErrors: Array<ApiErrorMessage | string> = [
+    { property: '500', message: 'Internal server error' },
+    { property: '400', message: 'Bad request error' },
+  ];
+
+  const errorsMessages: ValidationMessagesConfig = {
+    email: 'Invalid email',
+    required: 'Email is required',
+    minlength: 'Min length is 5',
+    maxlength: 'Min length is 10',
+  };
+
   const expectedControl = new FormControl('', [
     Validators.email,
-    Validators.min(10),
-    Validators.max(10),
     Validators.maxLength(10),
-    Validators.minLength(3),
+    Validators.minLength(5),
     Validators.required,
   ]);
 
-  beforeEach(async(() => {
-    TestBed.configureTestingModule({
-      imports: [ValidationMessagesModule],
-    })
-      .overrideComponent(ValidationMessagesComponent, {
-        add: {
-          providers: [
-            { provide: ControlContainer, useValue: {} },
-            { provide: MatFormField, useValue: {} },
-          ],
-        },
-      })
-      .compileComponents();
-  }));
-
-  beforeEach(() => {
-    fixture = TestBed.createComponent(ValidationMessagesComponent);
-    component = fixture.componentInstance;
-    component.control = expectedControl;
-    fixture.detectChanges();
+  it('should create', () => {
+    spectator = hostSetup(expectedControl);
+    expect(spectator.component).toBeTruthy();
   });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  it('should render multiple errors if multiple is set to true', () => {
+    spectator = hostSetup(
+      expectedControl,
+      undefined,
+      errorsMessages,
+      undefined,
+      true
+    );
+    const control = spectator.component.control;
+
+    control.setValue('test');
+    control.markAsDirty();
+
+    spectator.fixture.detectChanges();
+    expect(spectator.queryAll(baseSelector('validation-error'))).toHaveLength(
```

Comment:
if you're using `baseSelector('validation-error')` in many places, you can store it somewhere, like:
```ts
const selectors = {
  validationError: baseSelector('validation-error')
}
... 
spectator.query(selectors.validationError)

```
in your case it isn't much of a difference, but will help with longer names.